src/reader.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { Reader } from './reader'
+
+describe('Reader', () => {
+
+  type Cat = {
```

Comment:
I generally like to declare types outside any context. No need to put type declarations inside functions